### DISCUSSION TITLE
Building GIMP 2.10.36 AppImage on Open Build Service - https://build.opensuse.org/

### DISCUSSION THREAD

**Discussion by bert003** [AppImage/AppImageKit#1313] [General]
I simply love AppImages and would like to build GIMP 2.10.36 using the online service offered by OpenSUSE. Can someone please give me the steps to follow:

So far I:

1. Created a New Image and chose [Template for AppImage Based on Leap 15.2] https://build.opensuse.org/project/show/OBS:AppImage:Templates:Leap:15.2) but then when I specified GIMP as a package it downloaded their latest I guess (2.10.12).

2. I also tried to create a sub-project, however I got confused as to what repositories I need to add. I managed to download the `gimp-2.10.36.tar.bz2` from the gimp repository and uploaded it in the project. However when trying to build (I specified the commands to untar, configure, make etc) I got the error: `configure: error: *** A compiler with support for C++14 language features is required`. This means that I need gcc-6 I guess.

If I start a project from scratch, what repositories do I need to add? Also, can I import dependencies from any distribution offered (ex: Debian, Ubuntu etc) or just from OpenSuse in order to be able to build an appimage? Please help as I would really to accomplish this task.

Thanks in advance.

**Comment by probonopd** [comment]
Hello @bert003. Building a complex application like GIMP and deploying it to an AppImage is not a trivial task.

While I can't help you with the individual steps to do this on OBS (It's been a while since I last used it), I'd like to point you toward the existing work done regarding GIMP AppImage packaging by @aferrero2707:
https://github.com/aferrero2707/gimp-appimage

Overall, I think that reaching out to the GIMP developers and collaborating with them to produce an officially made and supported GIMP AppImage would be a sensible approach.

**Comment by bert003** [comment]
Thanks @probonopd - I will check the work done by aferrero2707 then.

From my end, I compiled GIMP and managed to install it in a directory named GIMP/usr. So I have the setup required I guess in the usr directory. Now when I create an AppImage, I am still getting the error that fonts cannot be found and even the required xml files cannot be found. It is like it looking for files in system's /usr and not the usr of the of the AppImage file (once it is extracted I presume).

Don't know what I am doing wrong at this point. Maybe I need to do some changes in the AppRun file?
